lib/snowflex/db_connection/protocol.ex

```diff
@@ -0,0 +1,86 @@
+defmodule Snowflex.DBConnection.Protocol do
+  use DBConnection
+
+  require Logger
+
+  alias Snowflex.DBConnection.{
+    Query,
+    Result,
+    Server
+  }
+
+  defstruct pid: nil, status: :idle, conn_opts: []
```

Comment:
May be worth adding `@enforce_keys [:pid]` to prevent an invalid `Protocol` from being constructed.